cli/lib/build.js

```diff
@@ -191,71 +164,149 @@ const checkModuleValid = async (execPath)=> {
   }
 }
 
-const checkRebuildModules = async () => {
-  if (fs.existsSync(path.resolve(publicDir, 'static'))) {
-    const { selectedModuleList } = await inquirer.prompt([
-      {
-        type: 'checkbox',
-        name: 'selectedModuleList',
-        message: 'Choose modules which need to rebuild',
-        choices: rebuildModules,
-      }
-    ]);
-    rebuildModules = selectedModuleList;
-  }
-}
-
-const buildAll =  async (enableSourceMap) => {  
+const buildModules = async (enableSourceMap, rebuildList) => {  
   const pList = [];
 
-  moduleList.forEach(item => {
+  const toBuildModules = rebuildList.length ? rebuildList : moduleList;
+  toBuildModules.forEach(item => {
     const { moduleName, moduleDir } = item;
+    // logInfo(`Building ${moduleName}`);
+    
+    let buildPromise = new Promise((resolve)=> {
+      const spinner = ora(`building ${moduleName}`).start();
+      exec('npm run build', { env: { ...process.env, enableSourceMap }, cwd: moduleDir, stdio: 'inherit' }, (error, stdout, stderr)=>{
+        if (error) {
+          logError(`build error: ${error}`);
+          process.exit(1);
+        } else {
+          logInfo(stderr);
+          logSuccess(`【${moduleName}】build successfully! [${stdout}]`);
+          resolve();
+          spinner.stop();
+        }
+      });
+    })
 
-    if (rebuildModules.includes(moduleName)) {
-      logInfo(`Building ${moduleName}`);
-  
-      let buildPromise = new Promise((resolve)=> {
-        exec('npm run build', { env: { ...process.env, enableSourceMap }, cwd: moduleDir, stdio: 'inherit' }, (error, stdout, stderr)=>{
-          if (error) {
-            logError(`build error: ${error}`);
-            process.exit(1);
-          } else {
-            logInfo(stderr);
-            logSuccess(`【${moduleName}】build successfully! [${stdout}]`);
-            resolve();
-          }
-        });
-      })
-  
-      pList.push(buildPromise);
-    }
+    pList.push(buildPromise);
   });
 
   await Promise.all(pList);
   logSuccess(`build successfully 😁!`);
 }
 
-module.exports = async (execPath) => {
-  try {   
-    await checkModuleValid(execPath);
+const stopDockerContainer = async () => {
+  await asyncExec('docker container stop erda-ui-for-build');
+  await asyncExec('docker rm erda-ui-for-build');
+}
+
+/**
+ * restore built content from an existing image
+ */
+const restoreFromDockerImage = async (image) => {
```

Comment:
I don't think is convenient to look for an existing image manually. It will be better if you can log recent image to local file every time you release image successfully.

Reply:
firstly, sometimes we don't want to release based on the most recent build.
secondly, if I log this into local file, how do we share it? commit it by git? then every build requires a extra PR
thirdly, if someone wants to take advantage of this feature，he should be pretty clear about what's going on. find out the correct version should be part of the job